packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -64,6 +68,7 @@ export class BackgroundScript {
     private loggingEnabled: boolean,
     @logger('BackgroundScript')
     private log: Logger,
+    private detectMultipleInstances: MultipleInstanceDetector,
```

Comment:
`detectMultipleInstances` 

A noun would be better. 
Consider `multipleInstanceDetector` or `multipleInstances`